### DISCUSSION TITLE
Adding table from dataflows error and no data in the table

### DISCUSSION THREAD

**Discussion by 10xape** [TabularEditor/TabularEditor3#1420] [Data Modeling]
Hi 

I am trying to connect additional table to a workspace model. The table is coming from the dataflow and is not incrementally refreshed. 
When I try to import the table I get the following errros and then the table appears with no columns or any data.

Appricate your help.

Matt
![image](https://github.com/user-attachments/assets/4fd55c3a-ae7e-4d78-9b9e-7b521503bb43)
![image](https://github.com/user-attachments/assets/1e264fa1-bd9d-4aa7-8706-9bcc0db55f75)

**Comment by otykier** [comment]
Did you import the table using TE3's Table Import Wizard? Could you share what the M expression on the partition generated by Tabular Editor looks like? (as usual, please blank out any sensitive information such as server names or passwords). Also, please confirm you're using the latest version [3.18.2](https://tabulareditor.com/downloads).

**Reply by 10xape** [reply]
Yes, I sued the wizard with connection to Power BI Dataflow, signed in multiplel times than oked and cancelled teh above error and teh table is added in this form of M code:

```
let
    Source = PowerBI.Dataflows(null),
    Workspace = Source{[workspaceId="452b4d13-ca25-46f2-XXXXXX"]}[Data],// PQ L2 Load Workspace
    Dataflow = Workspace{[dataflowId="a88bd2a5-2e36-4255-8820-XXXX"]}[Data], // go_security_PQv3.1
    Data = Dataflow{[entity="go_security_PQv3"]}[Data]
in
    Data
```

This is one of the most important functions for me to be able to add/remove table in the workspace mode  at any time. Thanks Matt

**Reply by otykier** [reply]
Thanks for clarifying. Could you please confirm that you're using v. 3.18.2?

Also, you said that you are prompted to sign in multiple times. Under normal circumstances, you should not be prompted to sign in more than twice (first time when you connect to the semantic model, second time when you use the Import Wizard to specify which Dataflow to import). Does that align with what you're seeing? If not, please provide more details and/or screenshots.

Here are a few other things you could try:

- Is your Power BI tenant located in one of the government clouds? If so, make sure to select the right API base URL under **Tools > Preferences > Power BI > Power BI Service**:

<img width="384" src="https://github.com/user-attachments/assets/5a8e869e-d1ad-432a-84a8-4a3c0a4f158f" />

- Enable **Use Analysis Services for change detection** under **Tools > Preferences > Schema Compare** (note, for this to work properly, you must ensure that credentials have been correctly specified under semantic models settings in the Power BI service:

<img width="384" src="https://github.com/user-attachments/assets/d389adb0-3fd8-451d-8ee5-eb4352faa22f" />

<img width="384" alt="image" src="https://github.com/user-attachments/assets/cbc52930-5c18-44b3-931e-4c8a0f1b6d61" />

**Reply by 10xape** [reply]
Hey, Yes I have 3.18.2 version, Yes I used teh wizard, no govermnet power bi service, this is my personal PPU Power BI license plan (company registered account). I have screenshotted the entire process, if I press ok instead of cancel it keeps coming back to sign in and worksapce selection and loops again.  Can we please resolve it asap as this is critical function for me. Thanks Matt
![image](https://github.com/user-attachments/assets/e53d3d9e-d73a-441e-ade9-1220f5205250)
![image](https://github.com/user-attachments/assets/242cf203-acf9-4141-9cd1-30cffb746be5)
![image](https://github.com/user-attachments/assets/a0bb75ac-120e-4e9f-9423-bab92637dea5)
![image](https://github.com/user-attachments/assets/f1b5cb0e-57bf-4ac8-b498-afabbaa2a3d8)
![image](https://github.com/user-attachments/assets/d955fa60-8e66-42ba-841a-146edd0b6700)
![image](https://github.com/user-attachments/assets/c0a15f03-441e-415c-8d4a-3e5ea4f5325a)
![image](https://github.com/user-attachments/assets/eae3a97b-125c-4ead-9797-1e844356166c)
![image](https://github.com/user-attachments/assets/7bd2ef7c-5f64-4816-806c-0c75f7eb48dd)
![image](https://github.com/user-attachments/assets/f8a3330d-0817-428d-9c95-94b892692628)

**Reply by otykier** [reply]
I was able to reproduce the issue when trying to import a dataflow entity without any columns, as in the screenshot below:

![image](https://github.com/user-attachments/assets/436a107a-7784-4e94-8a03-aff2ff4a6494)
![image](https://github.com/user-attachments/assets/9a347202-bc75-4a2d-b249-117a08f05eaa)
![image](https://github.com/user-attachments/assets/6de7fc6f-f28b-4128-9c60-8de49df5a851)

However, as soon as I choose a dataflow with a non-empty schema, it works fine in my testing:

![image](https://github.com/user-attachments/assets/55e4e829-cec2-4ccc-b020-a2eadbd5c2cb)

Could you confirm that the dataflow you're trying to load ("go_security_PQv3") does actually have a non-empty schema? Also, what does the **Choose source tables/views** page of the Import Tables wizard look like in your case (i.e. the page corresponding to the screenshot just above this paragraph)?

**Reply by otykier** [reply]
It would also be helpful if you could share the Dataflows definition JSON file with us (this is the JSON file that Tabular Editor attempts to parse, to determine the columns and datatypes of the tables in the Dataflow). You can share the file with us at support@tabulareditor.com

![image](https://github.com/user-attachments/assets/a2fb1736-2aea-415a-b7c3-71cbe0049910)

**Reply by 10xape** [reply]
hi it worksd now after the update, thank you

**Reply by otykier** [reply]
Glad to hear it. Out of curiosity, can you confirm that you are indeed importing a Dataflow table with no columns? If so, what is the use case of that?

**Comment by 10xape** [comment]
Hi Daniel

No , I am importing dataflows table with columns to add to the model.

Also is that possible to use tabular editor to launch the dataflows itself
in power bi service(not a model)?

Thanks

Matt
